src/app/Library/CrudPanel/CrudButton.php

```diff
@@ -35,6 +42,16 @@ public function __construct($name, $stack = null, $type = null, $content = null,
             extract($name);
         }
 
+        // check if position is a valid position
+        if ($position && ! ButtonPositionEnum::isValid($position)) {
+            throw new \Exception('CRUD Button position "'.$position.'" is not a valid position. Please use one of the following: '.implode(', ', ButtonPositionEnum::getValues()), 500);
+        }
+
+        // check if stack is a valid stack
+        if ($stack && ! ButtonStackEnum::isValid($stack)) {
+            throw new \Exception('CRUD Button stack "'.$stack.'" is not a valid stack. Please use one of the following: '.implode(', ', ButtonStackEnum::getValues()), 500);
+        }
+
```

Comment:
Hmm, I see you've added checks for the `position` and `stack`. 

PROs:
- better DX when devs add buttons to inexisting positions or stacks;

CONs:
- hardcodes the position and stack to what we have now, the usage inside ListOperation; if a developer wants to use CrudButtons in a custom operation... they can't... or they can, but they'd be forced to use only the stacks and positions we defined;

I think that CON is big enough to make us re-write this to remove the ENUMs and value-checking. For example, both next operations I had in mind to release:
- Calendar Operation (to manage events using FullCalendar)
- Kanban Operation (to manage issues/notes using boards)
would allow buttons, in a few places:
- Calendar - `before_calendar`, `top_left`, `top_right`, `after_calendar`, `item`
- Kanban Operation - `top`, `bottom`, `item`

With this change, a custom operation couldn't do that. So let's remove, please. 
Let me know if you disagree.

Reply:
I have too agree with you when you put it in that perspective. 

I will remove the enums 👍